### DISCUSSION TITLE
IOS Build Signing Fails and prevents the Build

### DISCUSSION THREAD

**Discussion by JonasKruegerCode** [codemagic-ci-cd/codemagic-docs#1970] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I build my app with the following IOS-Workflow.

![Screenshot 2023-08-14 143601](https://github.com/codemagic-ci-cd/codemagic-docs/assets/129793332/1792556e-8978-41ce-8441-757822c7334c)

The Apple Developer-Portal integration is active and the provisioning profile and Certifikate are sucessfully fetched

### Expected results

The App builds and is published to the App-Store

### Actual results

The Key and Profile are Imported but later on not found

![Screenshot 2023-08-14 142255](https://github.com/codemagic-ci-cd/codemagic-docs/assets/129793332/2d422cad-10fc-4608-bb10-61c6fa51eb41)

This makes the Build fail during the main build step.

![Screenshot 2023-08-14 142229](https://github.com/codemagic-ci-cd/codemagic-docs/assets/129793332/571c5496-3c24-4b36-8866-c105a7049c81)

### Build id (optional)

64da01d2660b2df2bda40fdb

**Comment by kulakarla** [comment]
Hi @JonasKrueger-techways

I understand you are using Code Signing Identities, having uploaded the certificate and profile to Codemagic.
If that is true, then remove the **Keychain** step from the build scripts and give the build another go.

Also, your bundle id-s are mismatching, your App Store bundle id is `io.techways.app`, yet your local projects bundle id is `com.example.techways`.

Try to fix these and if any issues persist, let me know.

**Reply by JonasKruegerCode** [reply]
Thanks for your fast reply. I am indeed using Code Signing Identities and have sucessfully uploaded them to Codemagic through the "fetch certifacte" and "fetch profiles" button. 

I removed the keychain step and the same error accured.

**Reply by kulakarla** [reply]
@JonasKrueger-techways 

Hey, I edited my reply - your bundle id-s are mismatching too.

**Reply by JonasKruegerCode** [reply]
thanks. Where can i see/edit my local projects bundle?

**Reply by kulakarla** [reply]
Inside `ios/Runner` there is a file `info.plist`. There, find this snippet:

```
<key>CFBundleIdentifier</key>
<string>com.your.packagename</string>
```

and change it to `io.techways.app`

Run `flutter clean` after changing them also.

**Reply by JonasKruegerCode** [reply]
changed it and error didn't change. The String was "$(PRODUCT_BUNDLE_IDENTIFIER)" before so maybe the fact that it pulled the wrong identifier points to another issue.

**Reply by kulakarla** [reply]
Share me the latest build ID, i will check.

**Reply by JonasKruegerCode** [reply]
Thanks. 64da2ed4f1844e098a9d9ba7

**Reply by kulakarla** [reply, marked as answer]
It is still the mismatch - if you have a Mac, change the bundle identifier in Xcode.


Otherwise, I advise you to do a change `ios/Runner.xcodeproj/project.pbxproj` , where you have the `PRODUCT_BUNDLE_IDENTIFIER`value. Change that also.

**Reply by JonasKruegerCode** [reply]
I changed `ios/Runner.xcodeproj/project.pbxproj` and it fixed the error.
Now the build fails at a later point:
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/129793332/c4542c55-a939-4934-815e-252c6b6a03d6)
It mentions the provisioning profile again so it might be connected. I will investigate further tomorrow and will reach out for additional help if needed. Thank you so much!

**Reply by kulakarla** [reply]
Hi @JonasKrueger-techways 

I see you selected an answer, did you get everything running smoothly now?

**Reply by JonasKruegerCode** [reply]
still not further than the error message above. I selected the awnser because it was the solution for the Problem mentioned in the beginning. I'm not sure if i should continue the discussion here or open a new issue for the Error during IPA-Building. 

I searched for further instnaces of the wrong bundle ID but couldnt find any. I found the line `"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";` in `ios/Runner.xcodeproj/project.pbxproj` and tried changing it to the Name of my identifier in the Apple developer portal. Didn't change anything.

Sadly I don't have access to a mac and find the whole Code-Signing process pretty confusing. I read most of Codemagics and Apples Documentation regarding Code-Signing but still don't really understand my issue.

**Reply by JonasKruegerCode** [reply]
my latest Build was 64db1c85393b4d44e16d1631

**Reply by kulakarla** [reply]
Yes, Apple Code Signing indeed can be tricky.
Can you try adding this `--export-options-plist=/Users/builder/export_options.plist` to your build script please.

And the change you described in your message before where its the `... = "iPhone Developer"`, please undo that.

This should fix it now.

**Reply by JonasKruegerCode** [reply]
That worked! Still had one Issue during publishing. It didn't detect the App Icons. But I should be able to fix that. Thank you so much for your help.

**Reply by kulakarla** [reply]
Yes, when building iOS apps with Flutter, in order to add the code signing identities, the `--export-options...` is vital since this essentially adds the signing identities.

Anyways, I will close the thread aswell. Glad you got everything working and happy building!
